staging/src/k8s.io/apimachinery/pkg/util/httpstream/httpstream.go

```diff
@@ -124,8 +124,8 @@ func Handshake(req *http.Request, w http.ResponseWriter, serverProtocols []strin
 	clientProtocols := req.Header[http.CanonicalHeaderKey(HeaderProtocolVersion)]
 	if len(clientProtocols) == 0 {
 		// Kube 1.0 clients didn't support subprotocol negotiation.
-		// TODO require clientProtocols once Kube 1.0 is no longer supported
-		return "", nil
+		// require clientProtocols once Kube 1.0 is no longer supported
+		return "", fmt.Errorf("clientProtocols is required")
```

Comment:
ideally this should be `errors.New`, but there are other instances of `fmt.Errorf` used instead in the code base.

is this assumption still valid 16 MINOR versions later?

/cc @ncdc

Reply:
if this goes live the comment:
> require clientProtocols once Kube 1.0 is no longer supported

is not needed.

Reply:
Subprotocol negotiation was added in 1.1. I imagine we can probably require it now. Do we need to go through an official deprecation-like notification period before doing so @liggitt?

Reply:
requiring client-protocols at this point is fine, though the error message should use the official header, and the corresponding serverProtocols change should be updated (along with unit tests)